### PR TITLE
Auto-generate settings reference documentation

### PR DESCRIPTION
**Pull request by desertaxle**
This PR adds a script to automatically generate reference documentation for all settings supported by Prefect. The generated documentation includes which environment variables are supported for each setting and the accessor path to use in a `prefect.toml` file. References to nested models include anchor tags for easier navigation. The settings reference and existing settings documentation are placed under the 'Develop' section in the documentation.

You can view a live version of the settings reference [here](https://prefect-bd373955-generate-settings-docs.mintlify.app/3.0/develop/settings-ref).

Screenshot 2024-10-28 at 7 38 19 PM

Related to https://github.com/PrefectHQ/prefect/issues/10697

A lot of descriptions for settings have room for improvement. I'll take a pass at improving those in another PR.

### CODE REVIEW

File: src/prefect/settings/base.py [hunk 1]
```diff
@@ -153,3 +154,31 @@ class PrefectSettingsConfigDict(SettingsConfigDict, total=False):
...
+        else:
+            env_vars.append(f"{model.model_config.get('env_prefix')}{property.upper()}")
+
+
+COMMON_CONFIG_DICT = {
+    "env_file": ".env",
+    "extra": "ignore",
+    "toml_file": "prefect.toml",
+    "json_schema_extra": add_environment_variables,
+}
```
Comment (desertaxle) [inline src/prefect/settings/base.py:184]: I found that a lot of this is repeated on all the models, so I consolidated it.

**Comment by desertaxle** [comment]
There's fixes for the failing tests in https://github.com/PrefectHQ/prefect/pull/15841

**Review by discdiver — COMMENTED** [review]
This is awesome! Thank you!

Could you please add a redirect to `mint.json` for `docs/3.0/manage/settings-and-profiles.mdx` → `docs/3.0/develop/settings-and-profiles.mdx`?

Is the `docs/3.0/api-ref/python/settings.mdx` file something that should be present? It looks duplicative of the Setting reference page and isn't in the nav menu. Is it doing something behind the scenes?

**Comment by desertaxle** [comment]
> Could you please add a redirect to `mint.json` for `docs/3.0/manage/settings-and-profiles.mdx` → `docs/3.0/develop/settings-and-profiles.mdx`?

Yup, added a redirect in [b3ddc51](https://github.com/PrefectHQ/prefect/pull/15830/commits/b3ddc51a47d7e8daa71d7ed776790edccb0fb377).

> Is the `docs/3.0/api-ref/python/settings.mdx` file something that should be present? It looks duplicative of the Setting reference page and isn't in the nav menu. Is it doing something behind the scenes?

Good catch! That was a leftover file that I forgot to remove, so it's gone now.

**Review by discdiver — APPROVED** [review]
lgtm

**Review by zzstoatzz — APPROVED** [review]

**Pull request merged**